npm/private/npm_package.bzl

```diff
@@ -256,19 +127,340 @@ def _impl(ctx):
+
+        include_transitive_sources: When True, `transitive_sources` from `JsInfo` providers in data targets are included in the list of available files to copy.
+
+        include_declarations: When True, `declarations` from `JsInfo` providers in data targets are included in the list of available files to copy.
+
+        include_transitive_declarations: When True, `transitive_declarations` from `JsInfo` providers in data targets are included in the list of available files to copy.
+
+        include_runfiles: When True, default runfiles from `srcs` targets are included in the list of available files to copy.
+
+            This may be needed in a few cases:
+
+            - to work-around issues with rules that don't provide everything needed in sources, transitive_sources, declarations & transitive_declarations
+            - to depend on the runfiles targets that don't use JsInfo
+
+            NB: The default value will be flipped to False in the next major release as runfiles are not needed in the general case
+            and adding them to the list of files available to copy can add noticeable overhead to the analysis phase in a large
+            repository with many npm_package targets.
+
+        hardlink: Controls when to use hardlinks to files instead of making copies.
+
+            Creating hardlinks is much faster than making copies of files with the caveat that
+            hardlinks share file permissions with their source.
+
+            Since Bazel removes write permissions on files in the output tree after an action completes,
+            hardlinks to source files are not recommended since write permissions will be inadvertently
+            removed from sources files.
+
+            - `auto`: hardlinks are used for generated files already in the output tree
+            - `off`: all files are copied
+            - `on`: hardlinks are used for all files (not recommended)
+
+        verbose: If true, prints out verbose logs to stdout
+
+        **kwargs: Additional attributes such as `tags` and `visibility`
+    """
+
+    files_target = None
+    if include_sources or include_transitive_sources or include_declarations or include_transitive_declarations or include_runfiles:
+        files_target = "{}_files".format(name)
+        _npm_package_files(
+            name = files_target,
+            srcs = srcs,
+            include_sources = include_sources,
+            include_transitive_sources = include_transitive_sources,
+            include_declarations = include_declarations,
+            include_transitive_declarations = include_transitive_declarations,
+            include_runfiles = include_runfiles,
+            # Always tag the target manual since we should only build it when the final target is built.
+            tags = kwargs.get("tags", []) + ["manual"],
+            # Always propagate the testonly attribute
+            testonly = kwargs.get("testonly", False),
+        )
+
+    if files_target:
```

Comment:
Can't this just merge with the `if` above?

Reply:
Hmm. Seems so. The code path changed and I didn't notice the duplicate